lib/aggregate/contracts/key_pair.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Aggregate
+  module Contracts
+    # Validates the specified hash key=>value combination can be an OriginKey => specified types
+    class KeyPair < ::Contracts::Builtin::CallableClass
+      def initialize(*vals)
+        @range = vals[0]
+        @key_types = vals[1]
+        @value_types = vals[2]
+        @error = nil
+      end
+
+      def valid?(hash)
+        new_hash = hash.dup.slice(*hash.keys[@range])
```

Comment:
This class does not have the `HashValueType` class as a base, so it doesn't validate that the variable `hash` is a hash.

If it doesn't throw an exception, then it is hash like enough, but it feels odd that you would have a validator that it is a hash, and a validator that assumes it is a hash and not either call the first within the second, or put both into the contract.